code/modules/security/genpop.dm

```diff
@@ -0,0 +1,721 @@
+	if(isobj(AM))
+		if(prob(50) && (machine_stat & BROKEN))
+			var/obj/O = AM
+			if(O.throwforce != 0)//don't want to let people spam tesla bolts, this way it will break after time
+				playsound(src, 'sound/magic/lightningshock.ogg', 100, 1, extrarange = 5)
+				tesla_zap(src, 3, 8000, TESLA_MOB_DAMAGE | TESLA_OBJ_DAMAGE | TESLA_MOB_STUN | TESLA_ALLOW_DUPLICATES) // Around 20 damage for humans
+	return ..()
+
+/obj/machinery/turnstile/welder_act(mob/living/user, obj/item/I)
+	//Shamelessly copied airlock code
+	. = TRUE
+	if(!I.tool_start_check(user, amount=0))
+		return
+	if(circuit == null && user.a_intent == INTENT_HARM)
+		var/obj/item/weldingtool/W = I
+		if(W.use_tool(src, user, 40, volume=50))
+			to_chat(user, "<span class='notice'>You start slicing off the bars of the [src]")
+			new /obj/item/stack/rods/ten(get_turf(src))
+			qdel(src)
+			return
+
+	if(obj_integrity >= max_integrity)
+		to_chat(user, "<span class='notice'>The turnstile doesn't need repairing.</span>")
+		return
+	user.visible_message("[user] is welding the turnstile.", \
+				"<span class='notice'>You begin repairing the turnstile...</span>", \
+				"<span class='italics'>You hear welding.</span>")
+	if(I.use_tool(src, user, 40, volume=50))
+		obj_integrity = max_integrity
+		set_machine_stat(machine_stat & ~BROKEN)
+		user.visible_message("[user.name] has repaired [src].", \
+							"<span class='notice'>You finish repairing the turnstile.</span>")
+		update_icon()
+		return
+
+/obj/machinery/turnstile/CanAllowThrough(atom/movable/mover, turf/target)
+	var/obj/item/card/id/id_card // used to check for prisoners trying to drag or piggyback others through the turnstile
+	. = ..()
+	if(. == TRUE)
+		return TRUE //Allow certain things declared with pass_flags_self through wihout side effects
+	if(machine_stat & BROKEN)
+		return FALSE
+	if(mover.dir == dir) //Always let people through in one direction.
+		return TRUE
+	var/allowed = allowed(mover)
+	//Ridden vehicles are blacklisted because they are items that have buckled mobs to them when this proc can only be called on mobs...
+	if(istype(mover, /obj/vehicle/ridden) && mover.buckled_mobs)
+		playsound(loc, 'sound/machines/buzz-sigh.ogg', 50)
+		say("ERROR. For security reasons, wheelchairs and other ridden devices are not allowed through the turnstile.")
+		flick("deny", src)
+		return FALSE
+	//Everyone with access can drag you out. Prisoners can't drag each other out
+	if(!allowed && mover.pulledby && ishuman(mover.pulledby))
+		var/mob/living/carbon/human/H = mover.pulledby
+		id_card = H.get_idcard(hand_first = TRUE)
+		if(ACCESS_PRISONER in id_card?.GetAccess())
+			return FALSE
+		else
+			allowed = allowed(mover.pulledby)
+	if(mover.buckled_mobs) //Piggyback rides aren't allowed because they could allow escape way too easily
+		for(var/mob/living/carbon/human/H in mover.buckled_mobs)
+			if(ACCESS_PRISONER in id_card?.GetAccess())
+				return FALSE
+			else
+				allowed = allowed(mover.buckled_mobs)
```

Comment:
ID card is only ever assigned in the above if block, meaning that if the thing being moved is not being pulled then ID card will always be null, which means that the second code block will be executed.
In theory this means that a prisoner that has no access could carry another prisoner out of the prison.
You may need to do `allowed = allowed && allowed(mover.buckled_mobs)`

Reply:
Also buckled mobs is a list, does `allowed()` allow lists as inputs?